src/value.cpp

```diff
@@ -862,14 +857,18 @@ namespace Exiv2 {
             static const char* ALPHA = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
             static const char* ALPHA_NUM = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
             
-            std::string::size_type pos = buf.find_first_of(' ');
-            lang = buf.substr(5, pos-5);
+            const std::string::size_type pos = buf.find_first_of(' ');
+            if (pos == std::string::npos) {
+                lang = buf.substr(5);
+            } else {
+                lang = buf.substr(5, pos-5);
+            }
             if (lang.empty()) throw Error(kerInvalidLangAltValue, buf);
             // Strip quotes (so you can also specify the language without quotes)
             if (lang[0] == '"') {
                 lang = lang.substr(1);
 
-                if (lang == "" || lang.find('"') != lang.length()-1)
+                if (lang.empty() || lang.find('"') != lang.length()-1)
```

Comment:
```suggestion
                if (lang.empty() || '"' != lang.back())
```

Reply:
That would change the behavior slightly, although your version might be what was actually intended. The old code also throws an exception if there's a `"` character in the middle of the string. Which behavior do you think we want?

Reply:
see comment above, we can't use `back()` anyway, so let's leave this as is.